core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -336,9 +336,9 @@ private[scheduler] object BlacklistTracker extends Logging {
    */
   def isBlacklistEnabled(conf: SparkConf): Boolean = {
     conf.get(config.BLACKLIST_ENABLED) match {
-      case Some(enabled) =>
+      case enabled =>
         enabled
-      case None =>
+      case _ =>
```

Comment:
This needs to be `getBoolean`. All of the other boolean properties have a default. It doesn't make sense here to have a third branch that deal with "not set" because a blacklist must either be enabled or not.